Mail/Components/AvatarView.swift

```diff
@@ -22,16 +22,34 @@ import MailResources
 import NukeUI
 import SwiftUI
 
-struct AvatarView: View, Equatable {
+struct AvatarView: View, Equatable, Identifiable {
     /// Optional as this view can be displayed from a context without a mailboxManager available
     let mailboxManager: MailboxManager?
 
-    let displayablePerson: CommonContact
+    var size: CGFloat
 
-    var size: CGFloat = 28
+    /// A view model for async loading of contacts
+    let viewModel: AvatarViewModel
+
+    /// Identifiable, linked to the contact displayed
+    var id: Int {
+        return viewModel.displayablePerson.id
+    }
+
+    /// Equatable
+    static func == (lhs: AvatarView, rhs: AvatarView) -> Bool {
+        lhs.id == rhs.id
+    }
+
+    init(mailboxManager: MailboxManager?, contactBuilder: CommonContactBuilder, size: CGFloat = 28) {
+        self.mailboxManager = mailboxManager
+        viewModel = AvatarViewModel(contactBuilder: contactBuilder)
```

Comment:
`AvatarViewModel` is not a StateObject but is `Observable` this won't work. 
Additionally, if you mark it as a StateObject, the AvatarViewModel won't be recreated every time. You will have to tie AvatarView's identity to the contact. (See my next comment)
[StateObject doc about initialization](https://developer.apple.com/documentation/swiftui/stateobject#Force-reinitialization-by-changing-view-identity)

Reply:
I did know that `StateObject` was to be used with `ObservableObject` but I did miss it. Thanks for the catch and the documentation.

Also none of my id/hash code is required, I played with it as it was not working for a too obvious reason and I froze.
With a proper { StateObject / ObservableObject / Published } for the view model, it works just fine.